framework/src/modules/base_module.ts

```diff
@@ -22,24 +22,37 @@ import {
 	AfterGenesisBlockApplyContext,
 	Reducers,
 	Actions,
+	BaseModuleDataAccess,
 } from '../types';
 import { BaseAsset } from './base_asset';
 
+interface Channel {
+	emit(name: string, data?: object): void;
+}
+
 export abstract class BaseModule {
 	public readonly config: GenesisConfig;
 	public transactionAssets: BaseAsset[] = [];
 	public reducers: Reducers = {};
 	public actions: Actions = {};
 	public events: string[] = [];
 	public accountSchema?: AccountSchema;
-
+	public channel!: Channel;
+	public dataAccess!: BaseModuleDataAccess;
```

Comment:
these 2 properties can be private?

Reply:
I think you mean `protected` as these props will be used in the extended classes

Reply:
or make a protected getter for both